components/bridge/Withdraw.tsx

```diff
@@ -1,65 +1,42 @@
 import React, { useEffect, useState } from "react";
 import { ethers } from "ethers";
-import { Box, Button, CircularProgress, TextField } from "@mui/material";
-import { Heading, SmallText } from "../../theme/StyledComponents";
 import { defineTxModal } from "../../utils/bridge/modal";
 import { useBlockchain } from "../../providers/BlockchainProvider";
 import { useCENNZApi } from "../../providers/CENNZApiProvider";
 import { useWallet } from "../../providers/SupportedWalletProvider";
 import TxModal from "./TxModal";
-import TokenPicker from "../shared/TokenPicker";
 import ErrorModal from "./ErrorModal";
-import { BridgeToken } from "../../types";
+import { BridgeToken, CennznetAccount } from "../../types";
+import ConnectWalletButton from "../shared/ConnectWalletButton";
 
-const ETH_CHAIN_ID = process.env.NEXT_PUBLIC_ETH_CHAIN_ID;
-
-const Withdraw: React.FC<{}> = () => {
-	const [token, setToken] = useState<BridgeToken>();
-	const [tokenBalance, setTokenBalance] = useState<Number>();
-	const [amount, setAmount] = useState("");
+const Withdraw: React.FC<{
+	token: BridgeToken;
+	amount: string;
+	selectedAccount: CennznetAccount;
+	disabled: boolean;
+	setEnoughBalance: Function;
+	setEstimatedFee: Function;
+}> = ({
+	token,
+	amount,
+	selectedAccount,
+	disabled,
+	setEnoughBalance,
+	setEstimatedFee,
+}) => {
 	const [modalOpen, setModalOpen] = useState(false);
 	const [errorModalOpen, setErrorModalOpen] = useState(false);
-	const [modalState, setModalState] = useState("");
+	const [modalState, _] = useState("");
```

Comment:
Same as above, why we don't need the `setModalState`?

Reply:
yeah don't think we need this state at all actually, modal state is stored in the `modal` state